packages/smart-wallet/src/smartWallet.js

```diff
@@ -366,6 +428,87 @@ export const prepareSmartWallet = (baggage, shared) => {
             },
           });
         },
+
+        provideBrandToPurses() {
+          const brandToPurses = provideLazy(
+            walletPurses,
+            this.facets.helper,
+            _k => {
+              /** @type {MapStore<Brand, PurseRecord[]>} */
+              const store = makeScalarBigMapStore('purses by brand', {
+                durable: true,
+              });
+              return store;
+            },
+          );
+          return brandToPurses;
+        },
+
+        /**
+         * Provide a purse given a NameHub of issuers and their
+         * brands.
+         *
+         * We current support only one NameHub, agoricNames, and
+         * hence one purse per brand. But we store an array of them
+         * to facilitate a transition decentralized introductions.
+         *
+         * @param {Brand} brand
+         * @param {ERef<NameHub>} known - namehub with brand, issuer branches
+         * @returns {Promise<Purse | undefined>} undefined if brand is not well known
+         */
+        async providePurseForKnownBrand(brand, known = shared.agoricNames) {
+          const brandToPurses = this.facets.helper.provideBrandToPurses();
+
+          if (brandToPurses.has(brand)) {
+            const purses = brandToPurses.get(brand);
+            if (purses.length > 0) {
+              // For now, we have just 1, so use it.
+              return purses[0].purse;
+            }
+          }
+
+          /**
+           * Introduce an issuer to this wallet.
+           *
+           * @param {Issuer} issuer
+           * @param {string} edgeName
+           */
+          const addIssuer = async (issuer, edgeName) => {
+            const [displayInfo, purse] = await Promise.all([
+              E(issuer).getDisplayInfo(),
+              E(issuer).makeEmptyPurse(),
+            ]);
+
+            // adopt edgeName as petname
+            // NOTE: for decentralized introductions, qualify edgename by nameHub petname
+            const petname = edgeName;
+            const assetInfo = { petname, brand, issuer, purse, displayInfo };
+            appendToStoredArray(brandToPurses, brand, assetInfo);
+            // NOTE: when we decentralize introduction of issuers,
+            // process queued payments for this brand.
+            return assetInfo;
+          };
+
+          const found = await namesOf(brand, E(known).lookup('brand'));
+          if (found.length === 0) {
+            return undefined;
+          }
+          const [edgeName] = found;
+          const issuer = await E(known).lookup('issuer', edgeName);
+
+          // Even though we rely on this nameHub, double-check
+          // that the issuer and the brand belong to each other.
+          try {
+            await mutualCheck(issuer, brand);
+          } catch (err) {
+            console.warn('issuer/brand mismatch', err);
+            return undefined;
+          }
```

Comment:
won't the error appear in the console upon throwing?

a failure of mutualCheck seems quite exceptional for a call to `providePurseForKnownBrand`. consider simplifying this to
```suggestion
          await assertMutual(issuer, brand);
```

Reply:
This code is called from `depositFacet.receive()`. So throwing here means dropping a payment on the floor.

Unless we try / catch in `receive()`. But assigning in a try/catch is ugly.

Reply:
from discussion: a match failure isn't so exceptional. Implies the brand isn't "known" (in a sane way) so returning undefined fits the description